optimum/deepsparse/modeling_diffusion.py

```diff
@@ -0,0 +1,527 @@
+            tokenizer (`CLIPTokenizer`):
+                Tokenizer of class
+                [CLIPTokenizer](https://huggingface.co/docs/transformers/v4.21.0/en/model_doc/clip#transformers.CLIPTokenizer).
+            scheduler (`Union[DDIMScheduler, PNDMScheduler, LMSDiscreteScheduler]`):
+                A scheduler to be used in combination with the U-NET component to denoise the encoded image latents.
+            feature_extractor (`Optional[CLIPFeatureExtractor]`, defaults to `None`):
+                A model extracting features from generated images to be used as inputs for the `safety_checker`
+            vae_encoder_session (`Optional[ort.InferenceSession]`, defaults to `None`):
+                The ONNX Runtime inference session associated to the VAE encoder.
+            model_save_dir (`Optional[str]`, defaults to `None`):
+                The directory under which the model exported to ONNX was saved.
+        """
+
+        self._internal_dict = config
+        self.model_save_dir = model_save_dir
+
+        self.is_static = static_shapes
+
+        self.vae_decoder_path = vae_decoder_path
+        self.vae_encoder_path = vae_encoder_path
+        self.text_encoder_path = text_encoder_path
+        self.unet_path = unet_path
+        self.text_encoder_2_path = text_encoder_2_path
+
+        self._device = device.upper()
+
+        self.compile(height=None, width=None)
+
+        self.tokenizer = tokenizer
+        self.tokenizer_2 = tokenizer_2
+        self.scheduler = scheduler
+        self.feature_extractor = feature_extractor
+        self.safety_checker = None
+
+        sub_models = {
+            DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER: self.text_encoder,
+            DIFFUSION_MODEL_UNET_SUBFOLDER: self.unet,
+            DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER: self.vae_decoder,
+            DIFFUSION_MODEL_VAE_ENCODER_SUBFOLDER: self.vae_encoder,
+            DIFFUSION_MODEL_TEXT_ENCODER_2_SUBFOLDER: self.text_encoder_2,
+        }
+
+        # Modify config to keep the resulting model compatible with diffusers pipelines
+        for name in sub_models.keys():
+            self._internal_dict[name] = (
+                ("diffusers", "OnnxRuntimeModel") if sub_models[name] is not None else (None, None)
+            )
+        self._internal_dict.pop("vae", None)
+
+    def compile(self, height, width, batch_size=1):
+        unet_batch_size = 2
+        # GET THIS FROM THE UNET CONFIG
+        # https://github.com/huggingface/optimum/blob/e6f83c550316a0859bc7183d8bdca6a85fffede8/optimum/pipelines/diffusers/pipeline_stable_diffusion.py#L290
+
+        # Create a dummy vae decoder in order to load config information
+        self.temp_vae_decoder = DeepSparseModelVaeDecoder(None, self)
+        if "block_out_channels" in self.temp_vae_decoder.config:
+            self.vae_scale_factor = 2 ** (len(self.temp_vae_decoder.config["block_out_channels"]) - 1)
+        else:
+            self.vae_scale_factor = 8
```

Comment:
can you also put this in the `is_static` section or is there other state to maintain?